File: frontend/src/types/booking.ts
export type BookingStatus = "pending" | "confirmed" | "cancelled";

export interface BookingCreate {
  room_id: string;
  user_id?: string;
  date: string; // ISO date string (YYYY-MM-DD)
  start_time?: string; // ISO time string (HH:MM:SS)
  end_time?: string; // ISO time string (HH:MM:SS)
  is_full_day: boolean;
  attendees: number;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  services: any[];
  total_price: number;
}

export interface BookingResponse extends BookingCreate {
  id: string;
  status: BookingStatus;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/booking.py
from pydantic import BaseModel, Field
from typing import List, Optional, Any, Literal
from datetime import date, time

BookingStatus = Literal['pending', 'confirmed', 'cancelled']

class BookingCreate(BaseModel):
    room_id: str
    user_id: Optional[str] = None
    date: date
    start_time: Optional[time] = None
    end_time: Optional[time] = None
    is_full_day: bool = False
    attendees: int
    services: List[Any] = []
    total_price: float

class BookingResponse(BookingCreate):
    id: str
    status: BookingStatus
    created_at: str
    updated_at: str